### PR TITLE
fix bug in patten search; enable set in http scope; enable work with …

### PR DESCRIPTION
**Pull request by zls0424**
* fix bug in patten search, use ngx_strlcasestrn instead of ngx_strcasestrn
* enable set in http scope
* enable work with userid module

### CODE REVIEW

**Comment by AirisX** [comment]
Hello @zls0424 ,

can you give an example of a case that leads to a bug that you try to fix in first commit?

**Comment by zls0424** [comment]
> Hello @zls0424 ,
> 
> can you give an example of a case that leads to a bug that you try to fix in first commit?

Hi AirisX,

when a cookie with name uuid is set in response header,  like 

`set-cookie: uuid=ZylrjlyJ2oAoAQksA1O5Ag==; expires=Thu, 31-Dec-37 23:55:55 GMT; path=/`

then cookie names like `uid`, `uuid`, `Ag` set with set_cookie_flag will all match  the cookie header line.

File: ngx_http_cookie_flag_filter_module.c [hunk 1]
```diff
@@ -235,57 +235,27 @@ ngx_http_cookie_flag_filter_init(ngx_conf_t *cf)
...
+        (u_char *)b, sizeof(b) - 1) == NULL) {                      \
+    p = ngx_pnalloc(r->pool, header->value.len + sizeof(b));        \
+    if (p == NULL) {                                                \
+      return NGX_ERROR;                                             \
+    }                                                               \
+    header->value.len = ngx_sprintf(p, "%V" b, &header->value) - p; \
+    header->value.data = p;                                         \
+    p[header->value.len] = '\0';                                    \
+  }                                                                 \
+} while(0)
```
Comment (edrandall) [inline ngx_http_cookie_flag_filter_module.c:252]: What is the purpose of the 'do..while' loop here, the original code has no such loop?
It's hard to follow how the original code maps into the macro refactoring.
Personally I would avoid large #define's and refactor the repeated-code-blocks into a function call, then add a unit test.
